Docs: Getting started edits

oddlittlebird opened the pull request:
Adjusted weights, moved some from mac.md to configuration.md, have some questions

**What this PR does / why we need it**:
- Adjusts Installation guide weights to make future organization easier
- Moves some configuration file from the Mac installation guide to the Configuration topic
- Makes minor edits to content as I start trying to understand what this section needs

**Which issue(s) this PR fixes**:
Starts on epic #19893

**Special notes for your reviewer**:

Please note the code comment questions.

oddlittlebird committed:
Getting started edits

Adjusted weights, moved some from mac.md to configuration.md, have some questions

oddlittlebird commented on docs/sources/installation/configuration.md:
```diff
@@ -17,7 +17,7 @@ specified in a `.ini` configuration file or specified using environment variable
 
 > **Note.** Grafana needs to be restarted for any configuration changes to take effect.
```
Why is this topic here rather than in Getting Started?

oddlittlebird commented on docs/sources/installation/mac.md:
```diff
@@ -62,24 +64,7 @@ brew services start grafana
 
 Default login and password `admin`/ `admin`
 
-
```
All removed content was moved to configuration.md. If it is generic rather than Mac-specific, then we should remove the "Mac"

oddlittlebird commented on docs/sources/installation/requirements.md:
```diff
@@ -47,29 +47,25 @@ Grafana supports the following databases:
-### Known issues
-
-#### Problem with logging in using Safari 12
+### Known issue: Problem with logging in using Safari 12
```
Why is this here rather than in Getting Started: Logging in to Grafana the first time?

oddlittlebird replied:
To clarify: Should I move it?

marcusolsson replied:
I think it's more related to the section on supported platforms, rather than Getting started. As in, these are the platforms we support, but we're also seeing this issue.

oddlittlebird replied:
@ivanahuckova , can you clarify where this should go?

ivanahuckova replied:
I was browsing the docs and honestly, I don't know. Maybe we could have a separate documentation for known issues? Or we can just add Note 4 with this information. Safari13 has been already released, so we can link the issue with Safari 12 and add a note to upgrade to Safari 13 or to Grafana 6.3.3.
![image](https://user-images.githubusercontent.com/30407135/67616174-e1ce5000-f7d5-11e9-9442-304b6b92d5a4.png)

oddlittlebird replied:
Thanks for pointing that out, @ivanahuckova. I think the Safari issue is probably covered under Note 1, so we can remove the specific bit about Safari 12. I will also clarify that we recommend users should upgrade their browsers to the latest version.

ivanahuckova replied:
@oddlittlebird yes, that's true! Great point.

oddlittlebird commented on docs/sources/installation/windows.md:
```diff
@@ -5,7 +5,7 @@ keywords = ["grafana", "configuration", "documentation", "windows"]
 [menu.docs]
 parent = "installation"
-weight = 3
+weight = 400
```
Line 49: Building what? This is a topic about installing, so where should this bit about building Grafana actually be?

oddlittlebird replied:
@torkelo , can you answer this?

oddlittlebird replied:
@ivanahuckova , can you answer this?

ivanahuckova replied:
My opinion - I think we shouldn't mix installation documentation with building from source documentation - and we should keep those 2 separate. Therefore I would remove this information from here. Also, I have found documentation related directly to [building from source](https://grafana.com/docs/project/building_from_source/), where is this exact information mentioned as well. However, I couldn't find this file in this repo - I think it makes sense to have it here and move it from the other location. 
What do you think? @torkelo @marcusolsson

marcusolsson replied:
I think we should differentiate between people who just want to build the latest version of Grafana, and people who are looking to develop Grafana. The latter should be covered in the [Developer guide](https://github.com/grafana/grafana/blob/master/contribute/developer-guide.md), but I think we should also consider having a stripped down guide for just building the binary. 

At this point though, everything in [Building from source](https://grafana.com/docs/project/building_from_source/) is covered in the Developer guide, so maybe we should replace it with a reference to the Developer guide for now?

oddlittlebird replied:
I like that plan, @marcusolsson. We don't want to confuse the laypeople, and it matches the way we've been organizing the contribution docs in the repo. 

Regarding the specific paragraph in windows.md, is that captured somewhere else already as well? Is it safe to delete it from windows.md?

marcusolsson replied:
I don't think it's mentioned elsewhere right now. Either we leave that part as is, or we move it to the Developer guide?

oddlittlebird replied:
Let's leave it here for now.

oddlittlebird commented on docs/sources/project/building_from_source.md:
```diff
@@ -3,7 +3,7 @@ title = "Building from source"
 [menu.docs]
 parent = "installation"
-weight = 5
+weight = 700
```
Why is this topic in the Projects folder rather than Installation?

torkelo replied:
because moving it means a URL change and need to add redirect, and we had a plan to restructure all the files and wanted to wait for that and do everything at once to avoid too many 301 redirects (SEO is important for the docs)

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -34,6 +34,8 @@ A common problem is forgetting to uncomment a line in the `custom.ini` (or `graf
 > `/etc/grafana/grafana.ini`. This path is specified in the Grafana
 > init.d script using `--config` file parameter.
 
+**Mac:** By default, the Configuration file is located at `/usr/local/etc/grafana/grafana.ini`.
```
```suggestion
**macOS:** By default, the configuration file is located at `/usr/local/etc/grafana/grafana.ini`.
```

Maybe we should restructure this section a bit. The default locations vary depending on how they were installed. 

If you downloaded the binaries yourself, the default is `$WORKING_DIR/conf/defaults.ini`.
If you installed using `deb` or `rpm` packages on Linux, the config file is in `/etc/grafana/grafana.ini`.
If you installed using `brew` on macOS, then `/usr/local/etc/grafana/grafana.ini`
If you installed on Windows, then it seems to be same as the binaries.

marefr replied:
Sounds good. If we don't want to have this on multiple places, maybe we can update each installation page and link to them? For example linking to https://grafana.com/docs/installation/rpm/#package-details for rpm.

oddlittlebird replied:
In the longer term, I am hoping to consolidate and simplify the Installation and Getting Started documentation, probably by having merging the sections and organizing them by system:
* Getting started with Windows
* Getting Started with MacOS
* And so on

ALL of this needs to be rewritten. This first PR is just trying to begin sorting content into the right places, and I am starting with the obvious, easy stuff. For example, configuration information should not be in the installation docs. Once everything is sorted into its proper "bucket," then we can start making decisions about how else to organize it.

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -78,9 +80,9 @@ system calls to get the machine name.
-command line in the init.d script or the systemd service file.
+Path to where Grafana stores the sqlite3 database (if used), file based sessions (if used), and other data.  This path is usually specified via command line in the init.d script or the systemd service file.
+
+**Mac:** The default SQLite database is located at `/usr/local/var/lib/grafana`
```
```suggestion
**macOS:** The default SQLite database is located at `/usr/local/var/lib/grafana`
```

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -89,14 +91,16 @@ How long temporary images in `data` directory should be kept. Defaults to: `24h`
+Path to where Grafana will store logs. This path is usually specified via command line in the init.d script or the systemd service file.  It can
+be overridden in the configuration file or in the default environment variable file.
+
+**Mac:** By default, the log file should be located at `/usr/local/var/log/grafana/grafana.log`.
```
```suggestion
**macOS:** By default, the log file should be located at `/usr/local/var/log/grafana/grafana.log`.
```

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -89,14 +91,16 @@ How long temporary images in `data` directory should be kept. Defaults to: `24h`
-be overridden in the configuration file or in the default environment variable
-file.
+Path to where Grafana will store logs. This path is usually specified via command line in the init.d script or the systemd service file.  It can
+be overridden in the configuration file or in the default environment variable file.
```
Looks like there's an unintentional line break here?

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -89,14 +91,16 @@ How long temporary images in `data` directory should be kept. Defaults to: `24h`
-Directory where grafana will automatically scan and look for plugins
+Directory where grafana will automatically scan and look for plugins. Manually or automatically install any plugins here.
+
+**Mac:** By default, the Mac plugin location is: `/usr/local/var/lib/grafana/plugins`.
```
```suggestion
**macOS:** By default, the Mac plugin location is: `/usr/local/var/lib/grafana/plugins`.
```

marcusolsson commented on docs/sources/installation/configuration.md:
```diff
@@ -89,14 +91,16 @@ How long temporary images in `data` directory should be kept. Defaults to: `24h`
 ### plugins
 
-Directory where grafana will automatically scan and look for plugins
+Directory where grafana will automatically scan and look for plugins. Manually or automatically install any plugins here.
```
```suggestion
Directory where Grafana will automatically scan and look for plugins. Manually or automatically install any plugins here.
```

marcusolsson commented on docs/sources/installation/mac.md:
```diff
@@ -5,12 +5,14 @@ keywords = ["grafana", "configuration", "documentation", "mac", "homebrew", "osx
 
 # Installing on Mac
 
+This page provides instructions to help you install Grafana on your Mac system. You can either install using homebrew or a binary tar file.
```
```suggestion
This page provides instructions to help you install Grafana on macOS. You can either install using homebrew or a binary tar file.
```

marcusolsson commented on docs/sources/installation/mac.md:
```diff
@@ -5,12 +5,14 @@ keywords = ["grafana", "configuration", "documentation", "mac", "homebrew", "osx
 +++
 
 
 # Installing on Mac
```
```suggestion
# Installing on macOS
```

marcusolsson commented on docs/sources/installation/mac.md:
```diff
@@ -90,11 +75,6 @@ To configure Grafana add a configuration file named `custom.ini` to the
 
-To run Grafana open your browser and go to http://localhost:3000/. 3000 is the default HTTP port that Grafana listens to if you haven't [configured a different port](/installation/configuration/#http-port).
-Then follow the instructions [here](/guides/getting_started/).
+Start Grafana by executing `./bin/grafana-server web`. The `grafana-server` binary needs the working directory to be the root install directory (where the binary and the `public` folder is located).
```
```suggestion
Start Grafana by executing `./bin/grafana-server`. The `grafana-server` binary needs the working directory to be the root install directory (where the binary and the `public` folder is located).
```

marcusolsson reviewed (changes requested):
Nice! Added a few suggestions, mainly on being consistent with macOS naming.

marefr commented on docs/sources/installation/configuration.md:
```diff
@@ -78,9 +80,9 @@ system calls to get the machine name.
-Path to where Grafana stores the sqlite3 database (if used), file based
-sessions (if used), and other data.  This path is usually specified via
-command line in the init.d script or the systemd service file.
+Path to where Grafana stores the sqlite3 database (if used), file based sessions (if used), and other data.  This path is usually specified via command line in the init.d script or the systemd service file.
```
Sessions is no longer in use

oddlittlebird replied:
So, what should I do with this content? 

Also, should "systemd" be "system.d"?

marcusolsson replied:
[systemd](https://github.com/systemd/systemd)

oddlittlebird replied:
Sounds good. I saw `init.d` and wondered if the systemd should be the same format.

marefr commented on docs/sources/installation/requirements.md:
```diff
@@ -7,23 +7,23 @@ type = "docs"
 - [Windows](/installation/windows)
 
-### Unsupported
+## Unsupported
```
This needs to be `Unsupported operating systems` since you've moved this to be a H2 instead of H3 - earlier the unsupported was a sub-header of `Operating systems`.

oddlittlebird replied:
Excellent, thanks for clarifying that!

marefr reviewed (dismissed):
Good job. Added some comments/request for changes.

oddlittlebird committed:
Update docs/sources/installation/configuration.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/configuration.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/configuration.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Edited configuration.md

oddlittlebird committed:
Merge branch 'installation-content-sorting1' of https://github.com/oddlittlebird/grafana into installation-content-sorting1

oddlittlebird committed:
Update docs/sources/installation/configuration.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/configuration.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/mac.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/mac.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update docs/sources/installation/mac.md

Co-Authored-By: Marcus Olsson <olsson.e.marcus@gmail.com>

oddlittlebird committed:
Update requirements.md

oddlittlebird committed:
Merge branch 'installation-content-sorting1' of https://github.com/oddlittlebird/grafana into installation-content-sorting1

oddlittlebird committed:
Update requirements.md

oddlittlebird committed:
Merge branch 'master' into installation-content-sorting1

oddlittlebird committed:
Update building_from_source.md

oddlittlebird commented:
I deleted the content from Building From Source and added a link to the Developer Guide in that file.

I see that there are still outstanding change requests from @marcusolsson  and @marefr, but everything is showing as either resolved or outdated in my view.

Is there anything else that needs to be done right now for this PR? I will be overhauling the Installation and Getting Started sections completely in a couple weeks.

marcusolsson reviewed (approved):
LGTM!

marcusolsson committed:
Merge branch 'master' into installation-content-sorting1
